IdfaIOS/CustomIdfa.m

```diff
@@ -0,0 +1,113 @@
+#import <AdSupport/AdSupport.h>
+#import <AppTrackingTransparency/AppTrackingTransparency.h>
+#import <CoreLocation/CoreLocation.h>
+
+//Please ensure this path points to the file 'Assets/Plugins/iOS/Amplitude/Amplitude.h'
+#import "../../Plugins/iOS/Amplitude/Amplitude.h"
+
+typedef NSString* (^AMPAdSupportBlock)(void);
+typedef NSDictionary* (^AMPLocationInfoBlock)(void);
+
+// Converts C style string to NSString
+// Note, this will produce C++ compiler/linker errors if this function is renamed to ToNSString or any function in Objective C++ files
+NSString* ToNSString2(const char* string)
+{
+    if (string)
+        return [NSString stringWithUTF8String: string];
+    else
+        return nil;
+}
+
+void setIdfaBlockInternal(const char* instanceName)
+{
+    NSLog(@"Setting idfa block");
+    AMPAdSupportBlock adSupportBlock = ^{
+        NSUUID *adUuid = [[ASIdentifierManager sharedManager] advertisingIdentifier];
+        NSString *result = [adUuid UUIDString];
+        return result;
+    };
+    [[Amplitude instanceWithName:ToNSString2(instanceName)] setAdSupportBlock:adSupportBlock];
+}
+
+@interface MyLocationManager : NSObject<CLLocationManagerDelegate> 
+{
+    CLLocationManager *locationManager;
```

Comment:
You can get rid of the curly bracket and just do 
`@property (nonatomic, strong) locationManager